compiler/sigmatch.nim

```diff
@@ -1045,7 +1045,7 @@ proc typeRelImpl(c: var TCandidate, f, aOrig: PType,
     return typeRel(c, f, lastSon(aOrig))
 
   if a.kind == tyGenericInst and
-      skipTypes(f, {tyVar, tyLent}).kind notin {
+      skipTypes(f, {tyVar, tyLent, tySink}).kind notin {
```

Comment:
Weird that ``tySink`` is the same as ``tyVar`` here when it IMHO should be treated like ``tyAlias``.